timesketch/lib/google_auth.py

```diff
@@ -38,11 +38,9 @@
 
 from timesketch.lib.definitions import HTTP_STATUS_CODE_OK
 
-
 CSRF_KEY = 'google_oauth2_csrf_token'
-AUTH_URI = 'https://accounts.google.com/o/oauth2/v2/auth'
-DISCOVERY_URL = 'https://accounts.google.com/.well-known/openid-configuration'
-
+AUTH_URI = current_app.config.get('GOOGLE_OIDC_AUTH_URI')
+DISCOVERY_URL = current_app.config.get('GOOGLE_OIDC_DISCOVERY_URL')
```

Comment:
this is what is making all the tests fail, what you are doing is to attempt to use the `current_app` in a constant at the module level, which is evaluated before the actual app is setup.

So what you need to do is a different approach...

```python
                                                                                                                                                 
DEFAULT_AUTH_URI = 'https://accounts.google.com/o/oauth2/v2/auth'                                                                                                                                                
DEFAULT_DISCOVERY_URL = (
    'https://accounts.google.com/.well-known/openid-configuration')
```

And then below you can do:

```python
    discovery_url = current_app.config.get(                                                                                                                                                                              
        'GOOGLE_OIDC_DISCOVERY_URL', DEFAULT_DISCOVERY_URL)                                                                                                                                                              
    try:                                                                                                                                                                                                                 
        resp = requests.get(discovery_url)             
```

That way,

a) you do not use the app before it is setup
b) for those that forget to update their config file (that is when you do a docker pull but didn't realize it required a change in the config file, things don't break
c) things just work ;)

Reply:
and the same applies for the other URL, that is do the `auth_uri = current_app....` instead of using the constant there.

Reply:
Noted. Thank you for your advice.